hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/main/java/org/apache/hadoop/yarn/server/federation/store/impl/ZookeeperFederationStateStore.java

```diff
@@ -203,53 +213,58 @@ public AddApplicationHomeSubClusterResponse addApplicationHomeSubCluster(
       FederationStateStoreUtils.logAndThrowStoreException(LOG, errMsg);
     }
     SubClusterId newSubClusterId =
-        request.getApplicationHomeSubCluster().getHomeSubCluster();
+         request.getApplicationHomeSubCluster().getHomeSubCluster();
     putApp(appId, newSubClusterId, true);
+
+    opDurations.addUpdateAppHomeSubClusterDuration(clock.getTime() - start);
     return UpdateApplicationHomeSubClusterResponse.newInstance();
   }
 
   @Override
   public GetApplicationHomeSubClusterResponse getApplicationHomeSubCluster(
-      GetApplicationHomeSubClusterRequest request) throws YarnException {
+          GetApplicationHomeSubClusterRequest request) throws YarnException {
 
+    long start = clock.getTime();
     FederationApplicationHomeSubClusterStoreInputValidator.validate(request);
     ApplicationId appId = request.getApplicationId();
     SubClusterId homeSubCluster = getApp(appId);
     if (homeSubCluster == null) {
       String errMsg = "Application " + appId + " does not exist";
       FederationStateStoreUtils.logAndThrowStoreException(LOG, errMsg);
     }
+    opDurations.addGetAppHomeSubClusterDuration(clock.getTime() - start);
     return GetApplicationHomeSubClusterResponse.newInstance(
```

Comment:
As this is all we do, would it make sense to have a:
```
GetApplicationHomeSubClusterResponse.newInstance(appId, homeSubCluster);
```
?

Reply:
Your suggestion is reasonable, I will modify the code.